circuit/types/mint_nft.go

```diff
@@ -78,17 +80,10 @@ func SetMintNftTxWitness(tx *MintNftTx) (witness MintNftTxConstraints) {
 	return witness
 }
 
-func ComputeHashFromMintNftTx(api API, tx MintNftTxConstraints, nonce Variable, expiredAt Variable, hFunc MiMC) (hashVal Variable) {
-	hFunc.Reset()
-	hFunc.Write(
-		PackInt64Variables(api, ChainId, tx.CreatorAccountIndex, nonce, expiredAt),
-		PackInt64Variables(api, tx.GasAccountIndex, tx.GasFeeAssetId, tx.GasFeeAssetAmount),
-		PackInt64Variables(api, tx.ToAccountIndex, tx.CreatorTreasuryRate, tx.CollectionId),
-		tx.ToAccountNameHash,
-		tx.NftContentHash,
-	)
-	hashVal = hFunc.Sum()
-	return hashVal
+func ComputeHashFromMintNftTx(api API, tx MintNftTxConstraints, nonce Variable, expiredAt Variable) (hashVal Variable) {
+	return poseidon.Poseidon(api, ChainId, TxTypeMintNft, tx.CreatorAccountIndex, nonce, expiredAt,
```

Comment:
I recall there was a #50 PR which optimized hash function computation and reduced number of constraints by using the fact that each variable takes 256 bits so several `int64` fields can be merged into one variable.
It looks that for Poseidon we're ignoring such optimization. I checked `Poseidon(...)` func but it looks that it translates each `int64` field into field element without trying to merge them.
Am I correct or do I miss something?

Reply:
The https://github.com/bnb-chain/zkbnb-crypto/pull/50 pr isn't correctly implemented, because the variable which consists of several `int64` fields may be bigger than `fr.Modulus`, this will cause potential hash collision attack. So I revert this optimization.